### PR TITLE
Added the Openstack Swift Provider.

### PR DESCRIPTION
**Pull request by varikin**
Currently just reads a file saved in a Swift object.

### CODE REVIEW

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackSwiftV1Provider.groovy [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  OpenstackIdentityProvider identityProvider
+
+  OpenstackSwiftV1Provider(OpenstackIdentityProvider identityProvider) {
+    this.identityProvider = identityProvider
+  }
+
+  @Override
+  public String readSwiftObject(String region, String container, String name) {
+    return handleRequest {
```
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackSwiftV1Provider.groovy]: remove return statement

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackSwiftV1Provider.groovy [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+public class OpenstackSwiftV1Provider implements OpenstackSwiftProvider, OpenstackRequestHandler, OpenstackIdentityAware {
+
+  OpenstackIdentityProvider identityProvider
+
+  OpenstackSwiftV1Provider(OpenstackIdentityProvider identityProvider) {
+    this.identityProvider = identityProvider
+  }
+
+  @Override
+  public String readSwiftObject(String region, String container, String name) {
```
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackSwiftV1Provider.groovy]: remove 'public'

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackSwiftV1Provider.groovy [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.openstack.client
+
+import org.openstack4j.model.common.DLPayload
+
+public class OpenstackSwiftV1Provider implements OpenstackSwiftProvider, OpenstackRequestHandler, OpenstackIdentityAware {
```
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackSwiftV1Provider.groovy]: remove 'public'

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackSwiftProvider.groovy [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+ */
+interface OpenstackSwiftProvider {
+
+  /**
+   * Returns the content of a Swift object.
+   * @param container the container that holds the object
+   * @param name the name the object within the container
+   * @return contents of the object
+   */
+  String readSwiftObject(String region, String container, String name)
```
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackSwiftProvider.groovy]: Do we want to be consistent with final parameters?

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackSwiftV1Provider.groovy [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+  OpenstackSwiftV1Provider(OpenstackIdentityProvider identityProvider) {
+    this.identityProvider = identityProvider
+  }
+
+  @Override
+  public String readSwiftObject(String region, String container, String name) {
+    return handleRequest {
+      DLPayload payload = getRegionClient(region).objectStorage().objects().download(container, name)
+      payload.getHttpResponse().getEntity(String)
```
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackSwiftV1Provider.groovy]: We should add null checks when getting the entity.

**Review by derekolk — CHANGES_REQUESTED** [review]

**Review by drmaas — CHANGES_REQUESTED** [review]

**Comment by varikin** [comment]
addressed comments, want to take another look?

**Review by derekolk — APPROVED** [review]

**Pull request closed without merging**